cmd/configurations.go

```diff
@@ -23,21 +24,32 @@ var ConfigurationsCmd = &cobra.Command{
 	Short: "List all Dapr configurations. Supported platforms: Kubernetes",
 	Run: func(cmd *cobra.Command, args []string) {
 		if kubernetesMode {
-			err := kubernetes.PrintConfigurations(configurationName, configurationOutputFormat)
+			if allNamespaces {
+				resourceNamespace = meta_v1.NamespaceAll
+			}
+			err := kubernetes.PrintConfigurations(configurationName, resourceNamespace, configurationOutputFormat)
 			if err != nil {
 				print.FailureStatusEvent(os.Stderr, err.Error())
 				os.Exit(1)
 			}
 		}
 	},
 	Example: `
-# List Kubernetes Dapr configurations
+# List default namespace Dapr configurations in Kubernetes mode
 dapr configurations -k
+
+# List define namespace Dapr configurations in Kubernetes mode
+dapr configurations -k -n default
+
+# List all namespaces Dapr configurations in Kubernetes mode
+dapr configurations -k --all-namespaces
 `,
 }
 
 func init() {
-	ConfigurationsCmd.Flags().StringVarP(&configurationName, "name", "n", "", "The configuration name to be printed (optional)")
+	ConfigurationsCmd.Flags().BoolVarP(&allNamespaces, "all-namespaces", "A", false, "If true, list all Dapr configurations in all namespaces")
+	ConfigurationsCmd.Flags().StringVarP(&configurationName, "name", "", "", "The configuration name to be printed (optional)")
+	ConfigurationsCmd.Flags().StringVarP(&resourceNamespace, "namespace", "n", "default", "List Define namespace configurations in a Kubernetes cluster")
```

Comment:
@artursouza @msfussell are we ok with changing the short hand flags for name and namespace ?? I think this is a breaking change also.

Reply:
Looking at this, there is no shorthand for "name" above. Given this is optional, I am ok with this breaking change for namespace, since this aligns with K8s, but could there be a shorthand for "name"?

Reply:
Can use "-N" for "name" for shorthand name?

Reply:
@msfussell, @artursouza Here based on the code, `-n` is changed from `name` to `namespace` which is a breaking change.

Reply:
cc @yaron2

Reply:
I am OK with this breaking change.

Reply:
In order to respect the deprecation policy, we should (in this PR), keep 100% backwards compatibility and give a warning that "-n" will mean namespace instead of name in the following release.